src/nix/config.cc

```diff
@@ -7,11 +7,31 @@
 
 using namespace nix;
 
-struct CmdShowConfig : Command, MixJSON
+struct CmdConfig : virtual NixMultiCommand
+{
+    CmdConfig() : MultiCommand(RegisterCommand::getCommandsFor({"config"}))
+    { }
+
+    std::string description() override
+    {
+        return "manipulate the Nix configuration";
+    }
+
+    Category category() override { return catUtility; }
+
+    void run() override
+    {
+        if (!command)
+            throw UsageError("'nix config' requires a sub-command.");
```

Comment:
Shouldn't we show the equivalent of `nix config --help` here instead? Not necessarily blocking, but it could be a quick fix with a tangible UX impact, and also set a good example.

Reply:
I followed what other commands are doing without thinking too much :stuck_out_tongue: 

I think failing here is good (because I might accidentally write that in a script, and I'd be glad if Nix tells me that it's a mistake), but _also_ showing the help (or at least the list of commands) could make a lot of sense

Reply:
I guess we can have an issue to factor this out :)